### ISSUE TITLE
Mounting a loop device: error

### ISSUE THREAD

**Issue by Fratacchione** [themaddoctor/linux-mybook-tools#16]
Hello Thomas, I wanted to thank you for your fantastic guide on "Mounting encrypted WD drives in Linux". It is really great and I think the effort you put into it is really admirable.

I'd like to ask you for some help, though, because unfortunately I'm encountering problems in the very last step of the guide: my father's WD-MyBook external drive used a JMicron chip (now it's fried) and after trying to use the HDD in a new, generic, HDD enclosure I think he did something wrong and let his Windows system mess with the MBR. I then did everything you suggested in detail, then went to the "Mounting with a loop device" section and followed those instructions too, but at the very last step, when you suggest to mount the loop device with :

sudo mkdir -p /mnt/wd
sudo mount /dev/loop2 /mnt/wd

(In my case it's "/dev/loop0")

I encounter this kind of error:

mount: /mnt/wd: wrong fs type, bad option, bad superblock on /dev/loop0, missing codepage or helper program, or other error.

How should I proceed? I did everything by the book, but why this last step is failing is beyond my comprehension, please help me ! :(

**Comment by themaddoctor** [comment]
I got your email and sent you a reply.


On 3/23/18, Fratacchione <notifications@github.com> wrote:
> Hello Thomas, I wanted to thank you for your fantastic guide on "Mounting
> encrypted WD drives in Linux". It is really great and I think the effort you
> put into it is really admirable.
>
> I'd like to ask you for some help, though, because unfortunately I'm
> encountering problems in the very last step of the guide: my father's
> WD-MyBook external drive used a JMicron chip (now it's fried) and after
> trying to use the HDD in a new, generic, HDD enclosure I think he did
> something wrong and let his Windows system mess with the MBR. I then did
> everything you suggested in detail, then went to the "Mounting with a loop
> device" section and followed those instructions too, but at the very last
> step, when you suggest to mount the loop device with :
>
> sudo mkdir -p /mnt/wd
> sudo mount /dev/loop2 /mnt/wd
>
> (In my case it's "/dev/loop0")
>
> I encounter this kind of error:
>
> mount: /mnt/wd: wrong fs type, bad option, bad superblock on /dev/loop0,
> missing codepage or helper program, or other error.
>
> How should I proceed? I did everything by the book, but why this last step
> is failing is beyond my comprehension, please help me ! :(
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/themaddoctor/linux-mybook-tools/issues/16


-- 
Thomas Kaeding

my public key: http://pgp.mit.edu/pks/lookup?op=get&search=0x81B0FCA32599BE6F

**Comment by techbio** [comment]
Came here having the same error with `sudo mount /dev/loop0 /mnt/wd`  except that I had completed the instructions for inic1607.

`mount: /mnt/wd: wrong fs type,...`

`dmesg | tail` gave me:

`FAT-fs (loop0): bogus logical sector size 65496`

Can you please help me to solve this? I am trying to access files on a 2TB MyBook and do not have drive space to use reallymine.

**Comment by themaddoctor** [comment]
I don't know if I can help you. I know how to set up the decryption, but I don't know how to repair FAT file systems.

**Comment by themaddoctor** [comment]
If this was a Mac disk, then try mounting with the option -t hfsplus

**Comment by klturi421** [comment]
@techbio or @Fratacchione - were either of you able to resolve your issues? I am experiencing the same thing and looking to see what I can do to get past that step.

**Comment by themaddoctor** [comment]
@klturi421 I told you here https://github.com/andlabs/reallymine/issues/69 that because you reformatted the drive after it was taken out of the WD enclosure, you CANNOT mount the decrypted image. You need to do data recovery on it.

**Comment by Erwinn94** [comment]
@themaddoctor Hi Thomas,

First of thank you for creating such a detailed .pdf guide. Just as @Fratacchione I went through everything in detail and went to the "Mounting with a loop device" section as I previously connected the drive to Windows OS. When mounting as suggested ('/dev/loop8' in my case) I get the following error:

mount: /mnt/wd: wrong fs type, bad option, bad superblock on /dev/loop8, missing codepage or helper program, or other error.

Is one of you (@themaddoctor & @Fratacchione) able to give possible solutions? Thank you in advance.

**Comment by Erwinn94** [comment]
Thank you for your reply @themaddoctor. It returns:

> root@ubuntu:~# sudo file -s /dev/loop7
/dev/loop7: data

**Comment by themaddoctor** [comment]
What does "sudo file -s /dev/mapper/wd" say?

**Comment by Erwinn94** [comment]
Returns:

> root@ubuntu:~# sudo file -s /dev/mapper/wd
/dev/mapper/wd: symbolic link to ../dm-2

**Comment by themaddoctor** [comment]
sudo file -s -L /dev/mapper/wd

**Comment by Erwinn94** [comment]
> root@ubuntu:~# sudo file -s -L /dev/mapper/wd
/dev/mapper/wd: data

Today I have also ran through the entire pdf again (incase of typing mistakes, which are unlikely due to copy-paste from pdf), but did not see different results.

**Comment by themaddoctor** [comment]
Fine. Dump the keyblock and first 2MB of the raw disk for me to look at.

**Comment by themaddoctor** [comment]
Yes, to your question. No, to the 2MB. You only dumped 2kB. Do:
dd if=/dev/sdX count=2048 of=start.bin
Then
7z a start.zip start.bin
and send "start.zip"

**Comment by Erwinn94** [comment]
[start.zip](https://github.com/themaddoctor/linux-mybook-tools/files/2280208/start.zip).

Thanks.

**Comment by themaddoctor** [comment]
Sorry. Other than the (not encrypted) MBR, there is nothing there but zeroes. No sign of encryption. No sign that the disk contains anything.

Try taking a sample from the middle of the disk. Use a skip number that is about the same as the number of kB of data that you think was on the disk. Sector are ½kB, so that should put you in the middle of the data.

**Comment by Erwinn94** [comment]
Yes, but I followed the document and am 100% sure I used the correct document. Currently I am trying to take a sample which is smaller than 10 MB but which includes data. When I made one of 66MB (zipped), I noticed in wordpad that the first lines states the one as posted before, then a lot of empty lines, and then a lot of data (?).

![naamloos](https://user-images.githubusercontent.com/42256003/43994878-a6002008-9da4-11e8-9257-0fec98b586bb.png)

**Comment by themaddoctor** [comment]
I can't understand your question.

What is the password?

**Comment by Erwinn94** [comment]
Can I send you an email with the password so I do not publicy publish the password?

**Comment by themaddoctor** [comment]
haecrapi@mytempemail.com

**Comment by Erwinn94** [comment]
Done, emailed.

**Comment by themaddoctor** [comment]
pls try again

**Comment by Erwinn94** [comment]
Still getting "address not found". I can send an email to the gmail account as stated at the top of the .pdf file?

**Comment by Erwinn94** [comment]
oh nvm, its withheld ofc.

**Comment by themaddoctor** [comment]
try sending me a private msg on github

**Comment by Erwinn94** [comment]
Don't think its possible to private msg on Github.

**Comment by themaddoctor** [comment]
pecad@bit-degree.com works

**Comment by Erwinn94** [comment]
Done, thanks.

**Comment by themaddoctor** [comment]
Use the script on this site called "inic1607e-extract-DEK-from-keyblock.sh":

./inic1607e-extract-DEK-from-keyblock.sh file-containing-kek.hex file-containing-keyblock.bin

**Comment by Erwinn94** [comment]
I think I succesfully entered the correct password as it shows a number instead of "decryption of keyblock failed". I will follow all the steps in the pdf document again. Thank you for your time so far, if it wasn't for your script I would never knew I entered the goddam wrong password.

**Comment by themaddoctor** [comment]
Is it the same number that you posted earlier?

**Comment by Erwinn94** [comment]
No, it is not 👍

**Comment by Erwinn94** [comment]
@themaddoctor It works! Thank you for retrieving my data. Your guide was great for a complete nooby as me who even entered the wrong decryption password, but with your script is was made possible. Once again, thank you!

**Comment by ng-anton** [comment]
HelloThomas,
Thank you for the detailed instruction on how to recover a WD My Book disk.
I have a malfunctioned 2TB WD My Book, removed the disk and mount it on Windows (my bad). I've followed your instructions, but when I do the last step to mount the loop I got an error like this :

ubuntu@ubuntu:~/Downloads$ sudo mount /dev/loop7 /mnt/wd
ntfs_attr_pread_i: ntfs_pread failed: Input/output error
Failed to calculate free MFT records: Input/output error
NTFS is either inconsistent, or there is a hardware fault, or it's a
SoftRAID/FakeRAID hardware. In the first case run chkdsk /f on Windows
then reboot into Windows twice. The usage of the /f parameter is very
important! If the device is a SoftRAID/FakeRAID then first activate
it and mount a different device under the /dev/mapper/ directory, (e.g.
/dev/mapper/nvidia_eahaabcc1). Please see the 'dmraid' documentation
for more details.

Do you have any idea how to recover from this error?

I think I already have the right DEK file, and I have double checked with the DEK from reallymine tool.

Thank you for your help.

**Comment by themaddoctor** [comment]
Please send me a copy of your keyblock, the dek, the password if any, and a dump of the first 2MB of the disk. You can make the dump with
dd if=/dev/sdX of=dump.bin count=4096
where X is replaced with the correct thing for your disk.

**Comment by ng-anton** [comment]
these are the files AYR. there's no password, so I used the default Pi code.

[dump.zip](https://github.com/themaddoctor/linux-mybook-tools/files/2623601/dump.zip)

**Comment by themaddoctor** [comment]
It decrypted OK for me, and I can see the MBR and the NTFS headers. Try mounting it the usual way (not with a loop device) because Windows did not corrupt the MBR in your case.

If you still get the same errors, then you might try to repair the NTFS filesystem. I think that ntfsprogs has a tool for that. You'll have to google it because I have never done such a thing. Or you might have to make an image of the decrypted disk (dd if=/dev/mapper/wd of=/dev/sdY) where Y is for a new empty disk to receive the image. Then put the cloned disk into a Windows machine.

**Comment by themaddoctor** [comment]
I may have spoken too soon. The partition table is empty. You may have to fix that, but don't let Windows do it, or it will overwrite the fist sector of the partition. And don't try to fix it until you have made the decrypted image.

**Comment by ng-anton** [comment]
> I may have spoken too soon. The partition table is empty. You may have to fix that, but don't let Windows do it, or it will overwrite the fist sector of the partition. And don't try to fix it until you have made the decrypted image.

So how can I fix the empty partition table?

**Comment by themaddoctor** [comment]
Don't do anything until you have a decrypted clone of the disk. Put the original disk in a safe place. Then use fdisk on the clone. Add a partition that starts at block 2048 and goes until the end of the disk. fdisk has built-in help.

**Comment by ng-anton** [comment]
> Don't do anything until you have a decrypted clone of the disk. Put the original disk in a safe place. Then use fdisk on the clone. Add a partition that starts at block 2048 and goes until the end of the disk. fdisk has built-in help.

Can I clone my 2TB disk to a 4TB disk? How should I create the partition if the new disk is 4TB?

**Comment by themaddoctor** [comment]
Do "cat /proc partitions" and write down what it says for your WD 2TB disk. Ask me later after you have cloned the disk.

**Comment by themaddoctor** [comment]
yes, thanks

**Comment by ng-anton** [comment]
> Do "cat /proc partitions" and write down what it says for your WD 2TB disk. Ask me later after you have cloned the disk.

Hi, Tom, I have cloned the disk, and 'cat /proc/partitions' return this :
major minor  #blocks  name

   7        0    1788456 loop0
   7        1      88632 loop1
   7        2     143400 loop2
   7        3       1652 loop3
   7        4      12504 loop4
   7        5      21488 loop5
   7        6       3332 loop6
   7        7 1953513560 loop7
   8        0  976762584 sda
   8        1     266240 sda1
   8        2      16384 sda2
   8        3  423581696 sda3
   8        4  503771136 sda4
   8        5    8227840 sda5
   8        6   26214400 sda6
   8        7    1024000 sda7
   8        8   12634112 sda8
   8        9    1024000 sda9
   8       16   30031872 sdb
   8       17   30030848 sdb1
   8       48 1953514584 sdd
 253        0 1953514584 dm-0
   8       32 3907017728 sdc
   8       33 3907015680 sdc1

I cloned the disk to /dev/sdc1 .

What should I do now?

**Comment by ng-anton** [comment]
> > Do "cat /proc partitions" and write down what it says for your WD 2TB disk. Ask me later after you have cloned the disk.
> 
> Hi, Tom, I have cloned the disk, and 'cat /proc/partitions' return this :
> major minor #blocks name
> 
> 7 0 1788456 loop0
> 7 1 88632 loop1
> 7 2 143400 loop2
> 7 3 1652 loop3
> 7 4 12504 loop4
> 7 5 21488 loop5
> 7 6 3332 loop6
> 7 7 1953513560 loop7
> 8 0 976762584 sda
> 8 1 266240 sda1
> 8 2 16384 sda2
> 8 3 423581696 sda3
> 8 4 503771136 sda4
> 8 5 8227840 sda5
> 8 6 26214400 sda6
> 8 7 1024000 sda7
> 8 8 12634112 sda8
> 8 9 1024000 sda9
> 8 16 30031872 sdb
> 8 17 30030848 sdb1
> 8 48 1953514584 sdd
> 253 0 1953514584 dm-0
> 8 32 3907017728 sdc
> 8 33 3907015680 sdc1
> 
> I cloned the disk to /dev/sdc1 .
> 
> What should I do now?

The source disk is in /dev/sdd .

**Comment by ng-anton** [comment]
> Use fdisk to make a partition starting at sector 2048 and ending at 3907029168

it means it will create 4TB partition instead of 2TB?

**Comment by themaddoctor** [comment]
No. Sectors are 512 bytes. Make the partition on the clone. Then put the clone in your windows machine and see if it can repair the filesystem.

I cannot help you further because I do not know anything about windows.

**Comment by ng-anton** [comment]
> No. Sectors are 512 bytes. Make the partition on the clone. Then put the clone in your windows machine and see if it can repair the filesystem.
> 
> I cannot help you further because I do not know anything about windows.

Should I create a DOS partition or a GPT partition?

**Comment by ng-anton** [comment]
Hi, Thomas. Thank you very much for your support. I can recover most of my files. I use testdisk to recover the NTFS partition that I lost on the drive. You're such a saviour. Thank you, my friend.

**Comment by themaddoctor** [comment]
You're welcome. Glad to hear it.
